### PR TITLE
[KUBOS-322] Updating Configs to Support Remote Update

### PR DESCRIPTION
**Pull request by CatFreed**
This PR depends on the [UBoot PR](https://github.com/kubostech/uboot/pull/3) being approved, since those changes are used to create the UBoot patch.

- Adding options for UBoot to support DFU utility.
- Adding partition mapping to add one partition for user data (separate from rootfs) and one partition to copy upgrade packages into.
- Adding U-Boot patch to support remote update
- Also turned on GDB, so we can debug user applications now

For dev use:
- Created default image tree source file (kpack.its) which describes the components that go into the upgrade package
- Created basic script (kubos-package.sh) which will create a rootfs image and then create the upgrade package file using the *.its, rootfs.img, and zImage files
  - The output file will be named kpack-{year}.{month}.{day}.itb with the intention that this file will then be uploaded to...somewhere...for users to download and then upgrade their systems with.  For now I'm using the date stamp as a way of versioning.

### CODE REVIEW

File: board/kubos/at91sam9g20isis/at91sam9g20isis_config [hunk 1]
```diff
@@ -12,20 +14,21 @@ CONFIG_CMD_MEMINFO=y
...
 # CONFIG_CMD_SOURCE is not set
 # CONFIG_CMD_SETEXPR is not set
-# CONFIG_CMD_NET is not set
 # CONFIG_CMD_NFS is not set
 CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
+CONFIG_DFU_TFTP=y
```
Comment (plauche) [inline board/kubos/at91sam9g20isis/at91sam9g20isis_config]: What uses the tftp config option?

File: kpack.its [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	#address-cells = <1>;
+
+	images {
+		zImage@1 {
+			data = /incbin/("./../buildroot-2016.11/output/images/zImage");
+			type = "firmware";
+			arch = "arm";
+			os = "linux";
+			compression = "none";
+			load = <00000000>;
```
Comment (plauche) [inline kpack.its]: Do these `load` and `entry` settings need unique values? Or are they relative?

File: board/kubos/at91sam9g20isis/at91sam9g20isis_config (hunk 1 above)
Comment (CatFreed) [inline board/kubos/at91sam9g20isis/at91sam9g20isis_config]: In uboot, update_kubos loads the new package, then calls update_tftp (the existing file upgrade utility), which then uses dfu_tftp

File: kpack.its (hunk 2 above)
Comment (CatFreed) [inline kpack.its]: No.  They're actually not used when the files are applied to SD card partitions.  I think it's for loading things into flash.  However, load (and maybe entry) is a required parameter for the existing upgrade utility.

**Pull request merged**